applications/RANSApplication/custom_strategies/bossak_scalar_scheme.h

```diff
@@ -0,0 +1,371 @@
+            << "     Relaxation factor           : " << this->mRelaxationFactor;
+
+        return msg.str();
+    }
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    struct BossakConstants {
+        double Alpha;
+        double Gamma;
+        double C0;
+        double C2;
+        double C3;
+    };
+
+    std::vector<LocalSystemVectorType> mSecondDerivativeValuesVectorOld;
+    std::vector<LocalSystemVectorType> mSecondDerivativeValuesVector;
+    std::vector<LocalSystemMatrixType> mMassMatrix;
+    std::vector<LocalSystemMatrixType> mDampingMatrix;
+
+    const double mAlphaBossak;
+
+    const Variable<double>& mrScalarVariable;
+    const Variable<double>& mrScalarRateVariable;
+    const Variable<double>& mrRelaxedScalarRateVariable;
+
+    BossakConstants mBossak;
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    static void CalculateBossakConstants(
+        BossakConstants& rBossakConstants,
+        const double Alpha,
+        const double DeltaTime)
+    {
+        TimeDiscretization::Bossak bossak(Alpha, 0.25, 0.5);
+        rBossakConstants.Alpha = bossak.GetAlphaM();
+        rBossakConstants.Gamma = bossak.GetGamma();
+
+        rBossakConstants.C0 =
+            (1.0 - rBossakConstants.Alpha) / (rBossakConstants.Gamma * DeltaTime);
+        rBossakConstants.C2 = 1.0 / (rBossakConstants.Gamma * DeltaTime);
+        rBossakConstants.C3 = (1.0 - rBossakConstants.Gamma) / rBossakConstants.Gamma;
+    }
+
+    template <class TItemType>
+    void AddMassMatrixToRHS(
+        TItemType& rItem,
+        LocalSystemVectorType& rRHS_Contribution,
+        const int ThreadId)
+    {
+        rItem.GetSecondDerivativesVector(mSecondDerivativeValuesVector[ThreadId], 0);
+        (mSecondDerivativeValuesVector[ThreadId]) *= (1.00 - mBossak.Alpha);
+        rItem.GetSecondDerivativesVector(mSecondDerivativeValuesVectorOld[ThreadId], 1);
+        noalias(mSecondDerivativeValuesVector[ThreadId]) +=
+            mBossak.Alpha * mSecondDerivativeValuesVectorOld[ThreadId];
+
+        noalias(rRHS_Contribution) -=
+            prod(mMassMatrix[ThreadId], mSecondDerivativeValuesVector[ThreadId]);
+    }
+
+    template <class TItemType>
+    void CalculateDynamicSystem(
+        TItemType& rItem,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        typename TItemType::EquationIdVectorType& rEquationIdVector,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY;
+
+        BaseType::CalculateSystemContributions(rItem, rLHS_Contribution, rRHS_Contribution,
+                                               rEquationIdVector, rCurrentProcessInfo);
+
+        const int k = OpenMPUtils::ThisThread();
+
+        rItem.CalculateMassMatrix(mMassMatrix[k], rCurrentProcessInfo);
+        // adding mass contribution to the dynamic stiffness
+        if (mMassMatrix[k].size1() != 0) // if M matrix declared
```

Comment:
Shouldn't we throw an error if the M matrix is not declared?

Reply:
we should not since this same methiod is used for conditions, and conditions by default return an empty mass matrix